helm/oauth2-proxy/templates/_helpers.tpl

```diff
@@ -78,11 +78,21 @@ Create the name of the service account to use
 {{- end -}}
 {{- end -}}
 
-
+{{/*
+Compute the redis url if not set explicitly.
+*/}}
 {{- define "oauth2-proxy.redisStandaloneUrl" -}}
+{{- $masterPort := .Values.redis.master.service.ports.redis -}}
 {{- if .Values.sessionStorage.redis.standalone.connectionUrl -}}
 {{ .Values.sessionStorage.redis.standalone.connectionUrl }}
+{{- else if .Values.redis.fullnameOverride -}}
+{{- printf "redis://%s-master:%.0f" (.Values.redis.fullnameOverride | trunc 63 | trimSuffix "-") $masterPort -}}
```

Comment:
You could probably build upon a mix of:
- How it is built in redis chart (using `common.names.fullname`)
- `common.names.dependency.fullname`

Like `printf "%s-master" (include "common.names.dependency.fullname" (dict "chartName" "redis" "chartValues" .Values.redis "context" $)`

in order not to have this whole logic hardcoded here.

Reply:
the problem is that the helper anicipates as context the chart itself. I didnt find out how to pass in the redis chart instead of the oauth2-proxy chart. Because using the parent chart is not correct.  The chart name that is used in this helper should be *redis*.

Reply:
Or is this the solution to the mentioned problem?

`dict "chartName" "redis" "chartValues" .Values.redis "context" $`

Because that would be amazing, that's right.

Reply:
I think I get it it, you are passing in a dict as context that *mocks* the redis chart. It contains the required values for the helper to function. Thats cool. I will make the required changes in a bit. 

Thanks for the feedback.